docs/how-to/migrate/forkid-7-to-9.md

```diff
@@ -0,0 +1,286 @@
+    ```
+
+4. Wait for the verified batch number to catch up to the trusted batch number:
+
+    ```sh
+    export ETH_RPC_URL="$(kurtosis port print cdk-v1 zkevm-node-rpc-001 http-rpc)"
+    cast rpc zkevm_batchNumber
+    cast rpc zkevm_verifiedBatchNumber
+    ```
+
+5. When those two numbers are the same, stop the services that are going to be upgraded:
+
+    ```sh
+    kurtosis service stop cdk-v1 zkevm-executor-pless-001
+    kurtosis service stop cdk-v1 zkevm-node-aggregator-001
+    kurtosis service stop cdk-v1 zkevm-node-eth-tx-manager-001
+    kurtosis service stop cdk-v1 zkevm-node-l2-gas-pricer-001
+    kurtosis service stop cdk-v1 zkevm-node-rpc-001
+    kurtosis service stop cdk-v1 zkevm-node-rpc-pless-001
+    kurtosis service stop cdk-v1 zkevm-node-sequence-sender-001
+    kurtosis service stop cdk-v1 zkevm-node-sequencer-001
+    kurtosis service stop cdk-v1 zkevm-node-synchronizer-001
+    kurtosis service stop cdk-v1 zkevm-node-synchronizer-pless-001
+    kurtosis service stop cdk-v1 zkevm-prover-001
+    ```
+
+## Smart contract calls
+
+1. Make the required smart contract calls:
+
+    ```sh
+    git clone git@github.com:0xPolygonHermez/zkevm-contracts.git
+    pushd zkevm-contracts/
+    git reset --hard a38e68b5466d1997cea8466dbd4fc8dacd4e11d8
+    npm i
+    printf "[profile.default]\nsrc = 'contracts'\nout = 'out'\nlibs = ['node_modules']\n" > foundry.toml
+    forge build
+    ```
+
+2. Deploy a new verifier. 
+
+    !!! tip
+        This step isn't strictly necessary but good to do because in some cases you need a new verifier contract.
+
+    ```sh
+    forge create --json \
+        --rpc-url "http://$(kurtosis port print cdk-v1 el-1-geth-lighthouse rpc)" \
+        --private-key 0x12d7de8621a77640c9241b2595ba78ce443d05e94090365ab3bb5e19df82c625 \
+        contracts/mocks/VerifierRollupHelperMock.sol:VerifierRollupHelperMock > verifier-out.json
+    ```
+
+3. Create a new rollup type for the upgraded network. In order to configure the file, we need a bunch of values from the l1 setup.
+
+    ```sh
+    kurtosis service exec cdk-v1 contracts-001 "cat /opt/zkevm/combined.json"
+    ```
```

Comment:
This step can be removed and the next step can be split into two steps:
1. Create the `PolygonValidiumStorageMigration` contract
2. Add a new rollup type to the rollup manager

Reply:
I'm unclear what's needed here @leovct